core/service.go

```diff
@@ -28,117 +32,195 @@ import (
 
 var _ services.Service = &Service{}
 
-// Service is a overhead layer that allows for communication between the runtime, BABE, and the p2p layer.
-// It deals with the validation of transactions and blocks by calling their respective validation functions
-// in the runtime.
+// Service is an overhead layer that allows communication between the runtime,
+// BABE session, and p2p service. It deals with the validation of transactions
+// and blocks by calling their respective validation functions in the runtime.
 type Service struct {
-	rt *runtime.Runtime
-	b  *babe.Session
-
-	msgRec  <-chan p2p.Message
-	msgSend chan<- p2p.Message
+	rt      *runtime.Runtime
+	bs      *babe.Session
+	bsRec   <-chan types.Block // receive blocks from BABE session
+	p2pRec  <-chan p2p.Message // receive messages from p2p service
+	p2pSend chan<- p2p.Message // send messages to p2p service
 }
 
-// NewService returns a Service that connects the runtime, BABE, and the p2p messages.
-func NewService(rt *runtime.Runtime, msgRec <-chan p2p.Message, msgSend chan<- p2p.Message) (*Service, error) {
-	b, err := babe.NewSession([32]byte{}, [64]byte{}, rt, msgSend)
+// NewService returns a new core service that connects the runtime, BABE session, and p2p service.
+func NewService(rt *runtime.Runtime, bsRec chan types.Block, p2pRec <-chan p2p.Message, p2pSend chan<- p2p.Message) (*Service, error) {
+
+	// create a new BABE session (bsRec becomes blockSend in babe)
+	bs, err := babe.NewSession([32]byte{}, [64]byte{}, rt, bsRec)
 	if err != nil {
 		return nil, err
 	}
 
-	return &Service{
+	s := &Service{
```

Comment:
if we are just returning a struct it's ok to put it all in the return line

Reply:
Ok, will put it all in the return line.